src/adapter/src/catalog/migrate.rs

```diff
@@ -51,7 +51,10 @@ pub(crate) async fn migrate(catalog: &mut Catalog) -> Result<(), anyhow::Error>
     };
     let mut tx = storage.transaction().await?;
     // First, do basic AST -> AST transformations.
-    rewrite_items(&mut tx, |_stmt| Ok(()))?;
+    rewrite_items(&mut tx, |stmt| {
+        subsource_type_option_rewrite(stmt);
```

Comment:
Ooh yay our first AST rewrite migration. Hope it works.

Reply:
The legacy upgrade tests make sure this happens successfully; fortunately, this is a well known pattern from the pre-cloud days so shouldn't be much of an issue.